frontend/apps/reader/modules/readersearch.lua

```diff
@@ -287,16 +289,22 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
     return true
 end
 
--- if regex==true use regular expression in pattern
--- if case == true or nil the search is case insensitive
+-- if regex == true, use regular expression in pattern
+-- if case == true or nil, the search is case insensitive
 function ReaderSearch:search(pattern, origin, regex, case_insensitive)
     logger.dbg("search pattern", pattern)
     local direction = self.direction
     local page = self.view.state.page
     if case_insensitive == nil then
         case_insensitive = true
     end
-    return self.ui.document:findText(pattern, origin, direction, case_insensitive, page, regex)
+    local retval, words_found = self.ui.document:findText(pattern, origin, direction, case_insensitive, page, regex, 2000)
+    if words_found and words_found > self.max_hits then
+        UIManager:show(Notification:new{
+            text =(T(_("Maximum hits reached: %1"), self.max_hits)),
+         })
```

Comment:
You feel it's necessary, with 2000 ? :)
The message might just be "Too many hits, some are not shown" - but that might be too long for a notification (at least, it's inviting to get long text for some translation languages).